### DISCUSSION TITLE
Create and provide a "flat" module description

### DISCUSSION THREAD

**Discussion by hendrikebbers** [support-and-care/maven-support-and-care#22] [Ideas for Funding]
While Maven already provides the pom.xml files that can be analysed to extract all (transitive) dependencies and build information, it would be a real benefit to have a file (created by the build) that contains a list of all used dependencies. That would make static analysis of a project even simpler since only file parsing is needed to get all dependencies, plugin information, and so on. Without that, static analyzers still need to know the exact pattern of how transitive dependencies and version clashes must be resolved.

I know that Gradle is currently working on exactly such a file format, and this may be another great opportunity to create a standard between build tools.

**Comment by slawekjaranowski** [comment]
What about SBOMs generated eg by https://github.com/CycloneDX/cyclonedx-maven-plugin
like: 
 - https://repo1.maven.org/maven2/org/apache/maven/shared/maven-dependency-analyzer/1.14.0/maven-dependency-analyzer-1.14.0-cyclonedx.json

**Reply by hendrikebbers** [reply]
yes, I was already thinking the same. But as far as I know SBOM does not contain all information that is needed: you can not define if a dependency is used at runtime/compiletime and you can not define if a dependency is used at test time.

**Reply by slawekjaranowski** [reply]
I did not more deeper in CycloneDX specification, but may be contain test and runtime artifacts as well.
